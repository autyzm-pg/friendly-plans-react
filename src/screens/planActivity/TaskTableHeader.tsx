import React, { SFC } from 'react';
import { StyleSheet, View } from 'react-native';

import { Icon, StyledText } from 'components';
import { i18n } from 'locale';
import { palette, typography } from 'styles';
import { CELL_DIMENSIONS } from './TaskTable';

export const TaskTableHeader: SFC = () => (
  <View style={styles.row}>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.NUMBER, alignItems: 'center' }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:number')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.CHECKBOX, alignItems: 'center' }]}>
      <Icon size={24} name="checkbox-blank-outline" color={palette.textDisabled} />
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.NAME }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:name')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.TYPE, alignItems: 'flex-start' }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:type')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.SECTION }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:section')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.TIME }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:time')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.DELETE, alignItems: 'flex-start' }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:delete')}</StyledText>
    </View>
    <View style={[styles.cell, { flex: CELL_DIMENSIONS.EDIT, alignItems: 'center' }]}>
      <StyledText style={styles.text}>{i18n.t('taskTable:edit')}</StyledText>
    </View>
  </View>
);

const styles = StyleSheet.create({
  row: {
    height: 32,
    alignSelf: 'stretch',
    flexDirection: 'row',
    paddingHorizontal: '6%',
    paddingVertical: 9,
  },
  cell: {
    alignSelf: 'stretch',
    justifyContent: 'center',
  },
  text: {
    color: palette.textDisabled,
    ...typography.overline,
  },
});
