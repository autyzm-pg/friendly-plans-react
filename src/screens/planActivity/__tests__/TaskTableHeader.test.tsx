import { shallow } from 'enzyme';
import React from 'react';

import { TaskTableHeader } from '../TaskTableHeader';

describe('TaskTableHeader', () => {
  it('renders correctly', () => {
    const taskTableHeader = shallow(<TaskTableHeader />);

    expect(taskTableHeader).toMatchSnapshot();
  });
});
