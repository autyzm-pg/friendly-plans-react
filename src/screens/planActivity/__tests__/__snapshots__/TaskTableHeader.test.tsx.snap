// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskTableHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flexDirection": "row",
      "height": 32,
      "paddingHorizontal": "6%",
      "paddingVertical": 9,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 0.3,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      #
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 0.3,
        },
      ]
    }
  >
    <Icon
      color="#9B9B9B"
      name="checkbox-blank-outline"
      size={24}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 1.7,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Name
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "flex-start",
          "flex": 0.5,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Type
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 0.5,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Parts
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "flex": 1.9,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Time
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "flex-start",
          "flex": 0.4,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Delete
    </StyledText>
  </View>
  <View
    style={
      Array [
        Object {
          "alignSelf": "stretch",
          "justifyContent": "center",
        },
        Object {
          "alignItems": "center",
          "flex": 0.6,
        },
      ]
    }
  >
    <StyledText
      style={
        Object {
          "color": "#9B9B9B",
          "fontFamily": "Rubik-Regular",
          "fontSize": 12,
          "letterSpacing": 0,
        }
      }
    >
      Edit
    </StyledText>
  </View>
</View>
`;
